se_apps/include/interface.h

```diff
@@ -109,526 +188,640 @@ static const float _focal_x = SceneK[0][0];
-				float v_v0_by_fy = (v - _v0) / _focal_y;
-
-				depthMap[u + v * _scenewidth] = depthMap[u + v * _scenewidth]
-						/ std::sqrt(
-								u_u0_by_fx * u_u0_by_fx
-										+ v_v0_by_fy * v_v0_by_fy + 1);
-
-			}
-		}
-		return index > 0;
-	}
+    }
+
+    inline uint2 getinputSize() {
+      return _size;
+    }
+
+    inline void restart() {
+      _frame = 0;
+    }
+
+    inline bool readNextDepthFrame(uchar3*, unsigned short int * depthMap) {
+
+      float* FloatdepthMap = (float*) malloc(
+          _size.x * _size.y * sizeof(float));
+      bool res = readNextDepthFrame(FloatdepthMap);
+
+      for (unsigned int i = 0; i < _size.x * _size.y; i++) {
+        depthMap[i] = FloatdepthMap[i] * 1000.0f;
+      }
+      free(FloatdepthMap);
+      return res;
+
+    }
+
+    inline bool readNextDepthFrame(float * depthMap) {
+
+      std::ostringstream filename;
+      get_next_frame();
+      filename << this->_dir << "/scene_00_" << std::setfill('0')
+        << std::setw(4) << _frame << ".depth";
+
+      std::ifstream source;
+      source.open(filename.str().c_str(), std::ios_base::in);
+
+      if (!source) {
+        std::cerr << "Can't open Data from " << filename.str().c_str()
+          << "!\n";
+        return 0;
+      }
+      uint index = 0;
+      while (source.good()) {
+        float d;
+        source >> d;
+        if (_scenewidth * _sceneheight <= index)
+          continue;
+        depthMap[index] = d;
+        index++;
+      }
+
+      for (int v = 0; v < _sceneheight; v++) {
+        for (int u = 0; u < _scenewidth; u++) {
+          float u_u0_by_fx = (u - _u0) / _focal_x;
+          float v_v0_by_fy = (v - _v0) / _focal_y;
+
+          depthMap[u + v * _scenewidth] = depthMap[u + v * _scenewidth]
+            / std::sqrt(
+                u_u0_by_fx * u_u0_by_fx
+                + v_v0_by_fy * v_v0_by_fy + 1);
+
+        }
+      }
+      return index > 0;
+    }
 
 };
 
+/**
+ * Reader for Slambench 1.0 datasets.
+ */
 class RawDepthReader: public DepthReader {
-private:
-	FILE* _pFile;
-	uint2 _size;
-
-public:
-	RawDepthReader(std::string filename, int fps, bool blocking_read) :
-			DepthReader(), _pFile(fopen(filename.c_str(), "rb")) {
-
-		size_t res = fread(&(_size), sizeof(_size), 1, _pFile);
-		cameraOpen = false;
-		cameraActive = false;
-		if (res != 1) {
-			std::cerr << "Invalid Raw file." << std::endl;
-
-		} else {
-			cameraOpen = true;
-			cameraActive = true;
-			_frame = -1;
-			_fps = fps;
-			_blocking_read = blocking_read;
-			fseeko(_pFile, 0, SEEK_SET);
-		}
-	};
-	ReaderType getType() {
-		return (READER_RAW);
-	}
-	inline bool readNextDepthFrame(uchar3* raw_rgb,
-			unsigned short int * depthMap) {
-
-		int total = 0;
-		int expected_size = 0;
-		unsigned int newImageSize[2];
-
-		get_next_frame();
-
-//		if (_frame < 2135)
-//		{
-//			_frame = 2135;
-//		}
-//
-//		std::cout << "Frame: " << _frame << std::endl;
+  private:
+    FILE* _pFile; /** Pointer to the open .raw file. */
```

Comment:
Better more explicative variable name than comment. They are internal it  won't be too much work, I hope

Reply:
Changed `_pFile` to `_rawFilePtr` and `_size` to `_inSize` in d0e0cf4dcde9e2be2904002f599c9a285638de8c.

Reply:
At this point the comments are irrelevant. Do you mind removing them?

Reply:
Comments removed in c079a8f12e3116bf057c7a108bc0588acbe7e1e3.